examples/hello_world/shared/src/counter.rs

```diff
@@ -0,0 +1,127 @@
+use crux_core::{render::Render, App};
+use crux_macros::Effect;
+use serde::{Deserialize, Serialize};
+
+#[derive(Serialize, Deserialize)]
+pub enum Event {
+    Increment,
+    Decrement,
+    Reset,
+}
+
+#[derive(Default)]
+pub struct Model {
+    count: isize,
+}
+
+#[derive(Default)]
+pub struct Hello;
+
+impl App for Hello {
+    type Model = Model;
+    type Event = Event;
+    type ViewModel = String;
+    type Capabilities = Capabilities;
+
+    fn update(&self, event: Self::Event, model: &mut Self::Model, caps: &Self::Capabilities) {
+        match event {
+            Event::Increment => model.count += 1,
+            Event::Decrement => model.count -= 1,
+            Event::Reset => model.count = 0,
+        };
+
+        caps.render.render();
+    }
+
+    fn view(&self, model: &Self::Model) -> Self::ViewModel {
+        format!("Count is: {}", model.count)
+    }
+}
+
+#[derive(Effect)]
+#[effect(app = "Hello")]
+pub struct Capabilities {
+    render: Render<Event>,
+}
```

Comment:
A really minor point, but should we declare this before it's used in the `App` impl?

Reply:
yeah, good point. I've fixed the ordering of a few things